### PR TITLE
Restructure

### PR DESCRIPTION
**Pull request by HiMyNameIsAdam**
By @EzIsFoglalt

### CODE REVIEW

File: source/sensors/sensor.py [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+    @classmethod
+    def configure_sensors(cls,config):
+        for ID,Value in config.items():
+            Type,Family = Value
+            cls.add_sensor(ID,Type,Family)
+    
+    @classmethod
+    def get_all_sensor_data(cls):
+        for _,_sensor in cls._instances.items():
+            print(_sensor._type.getState(_sensor))
```
Comment (EzIsFoglalt) [inline source/sensors/sensor.py]: Should not print

File: source/database/database.py [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
+Todo:
+"""
+import os
+
+import pymongo
+
+
+class Database:
+    """
+    This database handles the database interactions
```
Comment (HiMyNameIsAdam) [inline source/database/database.py]: This class*

File: source/database/database.py [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
+    URI = Address for the DB
+    """
+
+    URI = os.environ.get("MONGODB_URI") + "?retryWrites=false"
```
Comment (HiMyNameIsAdam) [inline source/database/database.py]: "?retryWrites=false" can be added to the env variable, so you don't have to append it here

File: source/database/sensordata.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    def __init__(self, name, stype, state, user_id, date=None, id_=None):
+        self.name = name
+        self.stype = stype
+        self.state = state
+        self.user_id = user_id
+        self.date = datetime.datetime.utcnow() if date is None else date
+        self.id_ = uuid.uuid4().hex if id_ is None else id_
+
+    def save_to_mongo(self):
```
Comment (HiMyNameIsAdam) [inline source/database/sensordata.py]: Should be renamed to save_to_db, so we have no knowledge about what type of db is being used

File: source/database/sensordata.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        Database.insert("test", self.json())
+
+    def json(self):
```
Comment (HiMyNameIsAdam) [inline source/database/sensordata.py]: return self.__dict__

File: source/database/user.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+""" Wrapped model above the documents stored in the
+    database inside the notification collection """
+import uuid
+import bcrypt
+from flask import session
+from common.database import DATABASE
```
Comment (HiMyNameIsAdam) [inline source/database/user.py]: import Database, not DATABASE
(Sorry :) )

File: source/database/user.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    def get_id(self):
+        return self._id
+    
+    @staticmethod
+    def get_current_user():
+        return User.get_by_email(session["email"])
+
+    @classmethod
+    def get_by_email(cls, email):
+        data = DATABASE.find_one("users", {"email": email})
```
Comment (HiMyNameIsAdam) [inline source/database/user.py]: Database

File: source/sensors/sensor_families/enOcean_sensor_family.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+class EnOcean:
+    """
+    This class handles the data exchange for the EnOcean sensors
+    Variables:
+        _baseurl (str): url to retrvie the raw data
+    """
+
+    _baseUrl = "http://192.168.1.10:8080/shapi/"
+
+    @classmethod
```
Comment (HiMyNameIsAdam) [inline source/sensors/sensor_families/enOcean_sensor_family.py]: Thous should be a @staticmethot, not a @classmethod

File: source/application/application.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    else:
+        print('User authentication failed, shutting down...')
+        return
+
+    Database.initialize()
+
+    while True:
+        time.sleep(time_out)
+        send_counter += 1
+        print(".", end="")
```
Comment (HiMyNameIsAdam) [inline source/application/application.py]: Caching missing!

**Pull request merged**